runtime/tchannel_client_test.go

```diff
@@ -73,3 +74,39 @@ func TestNilCallReferenceForLogger(t *testing.T) {
 	assert.True(t, resKey, "Client-Res-Header-header-key key not present")
 	assert.True(t, foo, "foo key not present")
 }
+
+func TestMaxAttempts(t *testing.T) {
+	methodName := map[string]string{
+		"methodKey": "methodValue",
+	}
+	tChannelClient := &TChannelClient{
+		maxAttempts: 2,
+		serviceName: "test",
+		methodNames: methodName,
+		timeout:     1,
+	}
+	ctx := context.TODO()
+	retryOpts := tchannel.RetryOptions{
+		MaxAttempts: 2,
+	}
+	contextBuilder := tchannel.NewContextBuilder(tChannelClient.timeout).SetParentContext(ctx).SetRetryOptions(&retryOpts)
+	maxAttempts := contextBuilder.RetryOptions.MaxAttempts
+	assert.Equal(t, 2, maxAttempts)
+}
+
+func TestMaxAttemptsDefault(t *testing.T) {
+	methodName := map[string]string{
+		"methodKey": "methodValue",
+	}
+	tChannelClient := &TChannelClient{
+		maxAttempts: 2,
+		serviceName: "test",
+		methodNames: methodName,
+		timeout:     1,
+	}
+	ctx := context.TODO()
+	retryOpts := tchannel.RetryOptions{}
+	contextBuilder := tchannel.NewContextBuilder(tChannelClient.timeout).SetParentContext(ctx).SetRetryOptions(&retryOpts)
+	maxAttempts := contextBuilder.RetryOptions.MaxAttempts
+	assert.Equal(t, maxAttempts, 0)
+}
```

Comment:
Please add one more test where we will create a new TChannelClient without passing maxAttemps and validating that maxAttempt value should be zero.
This will 100% validate that there will be no behaviour change with the existing implementation.

Reply:
Sure will add the required unit test.